.github/workflows/ci.yml

```
@@ -89,10 +93,10 @@ jobs:
           path: ~/.m2
           key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
           restore-keys: ${{ runner.os }}-m2
-      - name: Set up JDK11
-        uses: actions/setup-java@v1
+      - name: Set up the JDK
+        uses: actions/setup-java@v1.4.3
         with:
-          java-version: 11
+          java-version: ${{ matrix.java_reference }}
```

```suggestion
          java-version: 11
```

we cannot use the matrix variable here, because its scope is bound to the test job.
Maybe something like:
```suggestion
          java-version: ${{ jobs.test.matrix.java_reference }}
```
could also work but I havn't tested.
Or add an [output](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idoutputs) to the test job.